File: app/schemas/signals.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class TradeSignalResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    symbol: str
    direction: str
    strategy: str
    score: float
    status: str
    details: dict[str, Any]
    risk_approved: bool
    expires_at: datetime
    created_at: datetime


class SignalListResponse(BaseModel):
    items: list[TradeSignalResponse]
    count: int

File: frontend/src/api/types/signals.ts
export type SignalStatus = 'rejected' | 'review' | 'actionable' | 'candidate'
export type SignalDirection = 'bullish' | 'bearish' | 'neutral'

export interface TradeSignalResponse {
  id: string
  symbol: string
  direction: SignalDirection
  strategy: string
  score: number
  status: SignalStatus
  details: Record<string, unknown>
  risk_approved: boolean
  expires_at: string
  created_at: string
}

export interface SignalListResponse {
  items: TradeSignalResponse[]
  count: number
}

export interface TechnicalSignalResponse {
  symbol: string
  direction: SignalDirection
  confidence: number
  trend_score: number
  momentum_score: number
  volatility_score: number
  volume_score: number
  reasons: string[]
  warnings: string[]
}
